### PR TITLE
Replace CFR_ACC_ATOMIC with CFR_ACC_PERMITS_VALUE

### PR DESCRIPTION
**Pull request by hangshao0**
1. Remove CFR_ACC_ATOMIC and the corresponding testing code.
2. Replace CFR_ACC_ATOMIC with CFR_ACC_PERMITS_VALUE.
3. Add checks for ACC_PERMITS_VALUE and ACC_VALUE.

issue #14593

Signed-off-by: Hang Shao <hangshao@ca.ibm.com>

### CODE REVIEW

File: runtime/bcutil/cfreader.c [hunk 1]
```diff
@@ -2883,20 +2883,44 @@ j9bcutil_readClassFileBytes(J9PortLibrary *portLib,
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 	/* Currently value type is built on JDK19, so compare with JDK19 for now. Eventually it may need to compare to JDK20. */
 	if ((flags & BCT_MajorClassFileVersionMask) < BCT_JavaMajorVersionShifted(19)) {
-		classfile->accessFlags &= ~(CFR_ACC_VALUE_TYPE | CFR_ACC_PRIMITIVE_VALUE_TYPE | CFR_ACC_ATOMIC);
+		classfile->accessFlags &= ~(CFR_ACC_VALUE_TYPE | CFR_ACC_PRIMITIVE_VALUE_TYPE | CFR_ACC_PERMITS_VALUE);
 	}
 	/*
 	 * TODO This behaviour is based on the LW2 spec http://cr.openjdk.java.net/~fparain/L-world/LW2-JVMS-draft-20181009.pdf.
 	 * In the future the CFR_ACC_VALUE_TYPE class access bit will be replaced by a ValObject subtyping relationship. We will
```
[the remark this answers is not in the document]
Comment (hangshao0) [reply runtime/bcutil/cfreader.c]: Removed.

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity,extended xlinuxval jdknext

**Comment by hangshao0** [comment]
The compilation of VT build is now broken, see https://github.com/eclipse-openj9/openj9/pull/14700#issuecomment-1068356542

**Comment by gacholio** [comment]
https://github.com/eclipse-openj9/openj9/pull/14727 should fix the compile issue while we wait for #14700

**Comment by gacholio** [comment]
Jenkins test sanity,extended xlinuxval jdknext

**Pull request merged**